### ISSUE TITLE
Add "Initialise Component in Start()" context action

### ISSUE THREAD

**Issue by citizenmatt** [JetBrains/resharper-unity#356]
Given a field that derives from `Component` in a Unity class, e.g. `MonoBehaviour`, add a context action to initialise the value in the `Start()` method, in a similar way that current Rider/ReSharper will add code to initialise in the constructor. E.g. Given

```
public class MyClass : MonoBehaviour {
  private RigidBody _rigidBody;
}
```

Then there should be a quick fix on `_rigidBody` that will generate:

```
public class MyClass : MonoBehaviour {
  private RigidBody _rigidBody;
}

public void Start() {
  _rigidBody = GetComponent<RigidBody>();
}
```

**Comment by cmcpasserby** [comment]
would there be a option to choose, if it ends up in Start or Awake?

**Comment by CAD97** [comment]
👍 to option for `Start` vs `Awake`. As I've said before (#107), `Awake` is probably the better choice for this specific case especially.

General rule that seems to be roughly used: `Awake` for self-setup, `Start` when you need to observe other actors.

**Comment by citizenmatt** [comment]
Yep, this sounds like a good idea - two menu items, add to `Awake()`, add second to `Start()` (there's already an option to mark with `[SerializeField]`, and we'll aim to make that first).

Not sure what the difference is between "self-setup" and "observing", though - ownership?

**Comment by cmcpasserby** [comment]
While in Start you know that everything has called it's Awake already and it's safe to Start interacting with other objects. 

Awake would be the best place for this feature since is what should be used to setup stuff internal to your object and create references.

**Comment by citizenmatt** [comment]
See also [RIDER-15621](https://youtrack.jetbrains.com/issue/RIDER-15621)

**Comment by citizenmatt** [comment]
Closed with #1259
